src/main/java/io/tiledb/java/api/Query.java

```diff
@@ -467,6 +471,87 @@ public synchronized Query setBuffer(String attr, NativeArray buffer, long buffer
     return this;
   }
 
+  /**
+   * * Sets a NIO ByteBuffer
+   *
+   * @param attr The attribute
+   * @param bufferElements
+   * @return The NIO ByteBuffer
+   * @throws TileDBError
+   */
+  public synchronized ByteBuffer setBuffer(String attr, long bufferElements) throws TileDBError {
+    if (bufferElements <= 0) {
+      throw new TileDBError("Number of buffer elements must be >= 1");
+    }
+
+    Datatype dt;
+
+    try (ArraySchema schema = array.getSchema()) {
+      try (Domain domain = schema.getDomain()) {
+        if (domain.hasDimension(attr)) {
+          dt = domain.getDimension(attr).getType();
+        } else {
+          try (Attribute attribute = schema.getAttribute(attr)) {
+            dt = attribute.getType();
+          }
+        }
+      }
+    }
+
+    int size = Util.castLongToInt(bufferElements * dt.getNativeSize());
+
+    ByteBuffer buffer = ByteBuffer.allocateDirect(size);
+
+    // Set the byte order to the native system's native order
+    buffer.order(ByteOrder.nativeOrder());
+
+    this.byteBuffers_.put(attr, buffer);
```

Comment:
Let's call the `setBuffer(String attr, ByteBuffer buffer)` function from here, so we can avoid the code duplication once we create the ByteBuffer.

Reply:
done